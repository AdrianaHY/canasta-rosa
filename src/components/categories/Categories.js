import React, { Component } from 'react';

import './Categories.scss';

class Categories extends Component {
	render() {
		const { categories, handleProducts, isClicked} = this.props;
		return (
			<div className={isClicked ? 'show-list' : 'categories'}>
				<ul className="list-container">
					{categories.map((category) =>
						<li key={category.name}>
							<button className="category" onClick={() => handleProducts(category.slug)}>
								{category.name} <span className="chevron right" />
							</button>
						</li>
					)}
				</ul>
			</div>
		)
	}
}

export default Categories;