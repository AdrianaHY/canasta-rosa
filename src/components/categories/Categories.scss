@import '../../styles/_variables.scss';

.categories {
	display: none;
	flex-direction: column;
	width: 100%;
	padding: 0 $paddingMedium;

	@include breakpoint ('large') {
		padding: $paddingMedium $paddingLarge;
	}

	.show-list {
		display: flex;
		justify-content: space-between;


	}
}

