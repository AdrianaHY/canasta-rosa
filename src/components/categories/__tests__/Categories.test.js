import React from 'react';
import { shallow } from 'enzyme';
import Categories from '../Categories';

describe('components | categories | Categories', () => {
	it('Renders the component', () => {
		const props = {
			categories: [
				{
					name: 'category',
					slug: 'slug'
				}
			],
			isClicked: false,
			handleProducts: jest.fn()
		};

		const component = shallow(
			<Categories { ...props } />
		);

		expect(component).toMatchSnapshot();
	})
})