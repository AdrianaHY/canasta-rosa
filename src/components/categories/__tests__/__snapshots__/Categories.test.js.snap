// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | categories | Categories Renders the component 1`] = `ShallowWrapper {}`;
